### PR TITLE
Code Refactor

### PR DESCRIPTION
**Pull request by raymondxd**
- Update Eslint rules
- Fix lint errors
- Fix Vue errors in developer console

### CODE REVIEW

File: src/components/TreeMenu.vue
Comment (BradenM) [reply src/components/TreeMenu.vue:48]: Well -- Looking further, maybe I am just tired and am making that up haha.

Comment (raymondxd) [reply src/components/TreeMenu.vue:48]: @BradenM 
So using `v-for` and `:key` on the same element is not against **_vue:recommended_** rules.
One exception is when you use `v-for` with <template>. In that case, you need to use `:key` into its child elements directly 👇 
```
<template v-for="child in children">
  <div :key="`${child.id}" />
</template>
```
Thanks for asking 👍

File: src/components/WorksiteFilters.vue
Comment (tabiodun) [inline src/components/WorksiteFilters.vue:268]: Wyy `!!` instead of `Boolean`?

Comment (raymondxd) [reply src/components/WorksiteFilters.vue:268]: @tabiodun thanks for your comment
First of all, I was not sure what type of `option.value` is. I check the next code snippets and it was sth like string, I thought.
Based on this input, I think in order to convert string value into boolean, I simply used `!!` which will get true/false value from the strings.
Is `Boolean()` different from the above intention?

**Comment by tabiodun** [comment]
It’s the same. I was just wondering if you think it’s preferable to do this way rather than the explicit cast

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by raymondxd** [comment]
I just checked and yes, it's the same functionality.
If so, I will suit myself to the codestyle that you've been using to keep
consistency.
Thanks for pointing that out.

**Comment by tabiodun** [comment]
Awesome. Thanks. Also thanks for all these updates. I’ll look to approve once I get up.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by raymondxd** [comment]
My pleasure. Please leave more comments when you find any.

File: src/components/TreeMenu.vue
Comment (BradenM) [reply src/components/TreeMenu.vue:48]: :+1: Makes sense

File: src/layouts/Authenticated.vue
Comment (BradenM) [inline src/layouts/Authenticated.vue:22]: @raymondxd -- I think these key's were purposely being used in order to forcefully refresh the component... @tabiodun is this correct?

File: src/layouts/Authenticated.vue
Comment (BradenM) [inline src/layouts/Authenticated.vue:37]: Same here

File: src/pages/admin/AdminEvents.vue
Comment (BradenM) [inline src/pages/admin/AdminEvents.vue:160]: @raymondxd Oh you did end up camel-casing these?

**Review by BradenM — COMMENTED** [review]
@tabiodun @raymondxd LGTM -- except for maybe my comment above about the removal of the keys for component refresh.

File: src/layouts/Authenticated.vue
Comment (raymondxd) [reply src/layouts/Authenticated.vue:22]: Oops, sorry if that was added by intention. I removed these keys because key should be one of number or string. 
In the developer console, it was complaining that wrong type of key was used for them.
@tabiodun @BradenM So, if these keys are added by intention, I am good with adding them back but want to discuss the value to use for `:key` instead of using the user object directly.

File: src/layouts/Authenticated.vue
Comment (raymondxd) [reply src/layouts/Authenticated.vue:37]: Sorry if it was added by intention. I only checked that there was no `v-for` being used in the context so decided to remove it  as it's useless. 
@tabiodun @BradenM plz let me know if it's needed to be added back. waiting for you guys.

File: src/pages/admin/AdminEvents.vue
Comment (raymondxd) [reply src/pages/admin/AdminEvents.vue:160]: @BradenM Yes, I did.
I used es6 destructive code style to rename `user_badge` as `userBadge`, and made some following changes according to the variable name change

File: src/layouts/Authenticated.vue
Comment (BradenM) [reply src/layouts/Authenticated.vue:22]: Sure, I'll let @tabiodun handle this one since he wrote it :)

Comment (tabiodun) [reply src/layouts/Authenticated.vue:22]: Yeah they are for permission refreshes basically

File: src/layouts/Authenticated.vue
Comment (tabiodun) [reply src/layouts/Authenticated.vue:37]: Yeah. In places where a key exists without a `v-for` it's deliberate to help refresh the component

Comment (tabiodun) [reply src/layouts/Authenticated.vue:37]: So these will be needed

File: src/layouts/Authenticated.vue
Comment (raymondxd) [reply src/layouts/Authenticated.vue:22]: @tabiodun thanks for your confirmation.
If so, do we need to use user object for `key` there? What happens if we use `user.id` for `key`?

File: src/layouts/Authenticated.vue
Comment (raymondxd) [reply src/layouts/Authenticated.vue:37]: Ok, got it. I will add this back. Thank you for catching this out, @BradenM @tabiodun

File: src/layouts/Authenticated.vue
Comment (tabiodun) [reply src/layouts/Authenticated.vue:22]: The issues it that navbar changes based on the user permissions, not the user id. So what we really want is the key to be `user.premissions` but if the user isn't available, it will error

Comment (raymondxd) [reply src/layouts/Authenticated.vue:22]: @tabiodun I will revert it back as it was but plz keep in mind that it will cause errors on developer console as `user.permissions` is an object.
I think it will be better for you to think of changing `user.permissions` into a string to use for `:key`.

Comment (tabiodun) [reply src/layouts/Authenticated.vue:22]: We can do `JSON.stringify(user && user.premissions)` which will be a string maybe

Comment (raymondxd) [reply src/layouts/Authenticated.vue:22]: Great. let's do it then

Comment (raymondxd) [reply src/layouts/Authenticated.vue]: @tabiodun Now it's updated as we discussed. Plz check.

File: src/layouts/Authenticated.vue
Comment (raymondxd) [reply src/layouts/Authenticated.vue]: Now it's got key 🔙 . Please check.

**Review by BradenM — DISMISSED** [review]
LGTM

**Comment by BradenM** [comment]
While waiting for @tabiodun's final review -- will try and solve some of these conflicts.

**Comment by BradenM** [comment]
@raymondxd -- Oh, wow look at this. I must have missed it earlier. 

VueTypes _does_ still recommend [eslint-plugin-vue](https://dwightjack.github.io/vue-types/guide/installation.html#usage-with-eslint-plugin-vue) -- readding that should resolve the ~600 default prop warnings I got running `yarn lint`. 

What do you think?

**Review by BradenM — CHANGES_REQUESTED** [review]
See `vue-types` comment

**Comment by raymondxd** [comment]
@BradenM Thanks for pointing out about `vue-types` and `eslint-plugin-vue-types`.
I checked the documentation and yes, it was recommended to have `eslint-plugin-vue-types` when using `vue-types`.
I just added it back. Plz check.

**Review by BradenM — APPROVED** [review]
LGTM :tada:

**Pull request merged**
